compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -223,35 +224,25 @@ impl<'tcx> AbstractConst<'tcx> {
     }
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-struct WorkNode<'tcx> {
-    node: Node<'tcx>,
-    span: Span,
-    used: bool,
-}
-
 struct AbstractConstBuilder<'a, 'tcx> {
     tcx: TyCtxt<'tcx>,
-    body: &'a mir::Body<'tcx>,
+    body_id: thir::ExprId,
+    /// `Lrc` is used to avoid borrowck difficulties in `recurse_build`
+    body: Lrc<&'a thir::Thir<'tcx>>,
```

Comment:
what kind of borrowck difficulties are there, considering that this is an `Rc<&T>`? It seems a really weird to not use a reference here :thinking:

Reply:
apparently no borrow check difficulties, yay